go-fips-1.20.yaml

```diff
@@ -27,11 +27,11 @@ environment:
       - openssl-dev
 
 pipeline:
-  - uses: fetch
+  - uses: git-checkout
     with:
-      uri: https://go.dev/dl/go${{package.version}}.src.tar.gz
-      expected-sha256: 1aef321a0e3e38b7e91d2d7eb64040666cabdcc77d383de3c9522d0d69b67f4e
-      strip-components: 0
+      repository: https://github.com/golang/go.git
+      tag: go${{package.version}}
+      expected-commit: 90a870f1dc49bfcc6ffe95f80fbaf21875198e7a
 
   - working-directory: /home/build/go
```

Comment:
So go/ dir got dropped from git-checkout that was in fetch before;
And it was droppped from build;
And this correctly fails with old melange;
And it passed with new melange..... because we likely have a regression in working-directory handling in a sub-pipeline.

Reply:
cc @smoser @jonjohnsonjr

Reply:
git bisect says that https://github.com/chainguard-dev/melange/commit/ed837f56957f8a4bbe119ae26ebba296f440e308 Separate compilation from execution is the regression, with sub-pipelines working-directory being ignored.

CC: @jonjohnsonjr

Reply:
plot thickens, it seems to only affect, nested sub-piplines i.e. those that are created by "uses", as if working directory inheritence is not done 3 times down:

```yaml
  - working-directory: /home/build/go
    pipeline:
      - runs: |
          echo $(pwd) should be /home/build/go # is good and runs in /home/build/go
      - uses: patch # is bad and runs in /home/build
        with:
          patches: /home/build/000-initial-setup.patch
```